src/components/memorygame/MemoryGameBoard/index.jsx

```diff
@@ -0,0 +1,128 @@
+  //    description:"",
+  //    link:"",
+  //    phrase:""
+  // }
+  const [cardStates, setCardStates] = useState(
+    cardInitialStateGenerator(cardsDataArr, NUM_REQUIRED_MATCHES)
+  );
+  // selectedCards is an array of clicked card states(objects).
+  // selectedCards array hold only "NUM_REQUIRED_MATCHES" amount of card states
+  const [selectedCards, setSelectedCards] = useState([]);
+  // Checks if two cards have a matching image.
+  const areCardsMatching = (card, firstCard) => {
+    return card.img.imgKey === firstCard.img.imgKey;
+  };
+  // Triggered when user clicks on a card(selectedCards change)
+  // checks selected cards to see if they are a match
+  useEffect(() => {
+    // prevents any action to be taken if there isn't enough clicked card
+    if (selectedCards.length !== NUM_REQUIRED_MATCHES) {
+      return;
+    }
+    const isThereMatch = selectedCards.every((card) => {
+      return areCardsMatching(card, ...selectedCards);
+    });
+    if (isThereMatch) {
+      // When isThereMatched condition is provided all cards are same.
+      // Because selectedCards gets resetted after every successful or unsuccessful match.
+      selectedCards.forEach((matchedCard) => {
+        // isMatched property of each card state in selectedCard get changed into true
+        // this makes success card to appear instead of flippable card
+        changeCardStatePropertyToOpposite(
+          "isMatched",
+          matchedCard,
+          cardStates,
+          setCardStates
+        );
+      });
+      // matchedCards hold successfuly matched card states and uses them to display facts
+      // to prevent fact duplication only 1 matched card should be passed into matchedCards
+      setMatchedCards((prevState) => [...prevState, selectedCards[0]]);
+    } else {
+      selectedCards.forEach((unmatchedCard) => {
+        // before this step handleClick made isFlipped property of all selected cards true.
+        // since match failed now,
+        // isMatched property of each card state in selectedCard get changed into false
+        // This makes cards to flip back
+        changeCardStatePropertyToOpposite(
+          "isFlipped",
+          unmatchedCard,
+          cardStates,
+          setCardStates
+        );
+      });
+    }
+    setSelectedCards([]);
+  }, [selectedCards]);
+
+  // displays a modal when victory condition achieved
+  useEffect(() => {
+    const MAX_MATCHES = cardStates.length / NUM_REQUIRED_MATCHES;
+    if (matchedCards.length === MAX_MATCHES) {
+      setIsGameCompleted(true);
+      Modal.success({
+        content: "You have completed the game!",
+      });
+    }
+  }, [matchedCards]);
+
+  return (
+    <div>
+      <Row align="middle" className="howToPlayPopoverAndScrollToFactsRow">
+        <Col span={5}>
+          <ScrollToFacts />
+        </Col>
+        <Col span={2} push={17}>
+          <HowToPlayPopover />
+        </Col>
+      </Row>
+      <Row>
+        <GameCards
```

Comment:
unless you completely rewrote the code, notice how the diff on the left side is empty and its saying you added 128 lines.

you want to use git mv to move MemoryGameCards/idnex.js to MemoryGameBoard/index.js, then modify it.

imagine you are me. i am wondering whether the code was completely rewritten or whether you rewrote just parts of it. but now i have to rereview the whole thing.

https://stackoverflow.com/questions/433111/how-to-make-git-mark-a-deleted-and-a-new-file-as-a-file-move

basically you shouldnt be deleting and then copy-pasting to a new file, you should be doing `git mv` followed by `git commit` and that should make the diff show up correctly (unless github is strange about it). but give it a try.

Reply:
what i should see is a rename + a diff of how the code was changed. otherwise i'm rereviewing the whole code.